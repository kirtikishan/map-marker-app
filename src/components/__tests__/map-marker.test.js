import React from "react";
import { shallow } from "enzyme";
import { Provider } from "react-redux";

import configureMockStore from "redux-mock-store";
import MapMarker from "../map-marker";

const mockStore = configureMockStore();
const store = mockStore({});

describe("MapMarker", () => {
  it("should render map marker", () => {
    const mapObject = {
      markerProps: {
        position: [500, 65],
      },
      mapData: {
        show: true,
        modalData: {
          name: "Black Crystal",
          type: "small town",
          population: 1385,
          wealth: 29842,
          authority: "constable",
          numGuards: 13,
          position: [244, 141],
        },
      },
    };

    const tree = shallow(
      <Provider store={store}>
        <MapMarker props={mapObject} />
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });
});
