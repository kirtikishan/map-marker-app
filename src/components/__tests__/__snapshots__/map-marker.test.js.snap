// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapMarker should render map marker 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(Marker)
    props={
      Object {
        "mapData": Object {
          "modalData": Object {
            "authority": "constable",
            "name": "Black Crystal",
            "numGuards": 13,
            "population": 1385,
            "position": Array [
              244,
              141,
            ],
            "type": "small town",
            "wealth": 29842,
          },
          "show": true,
        },
        "markerProps": Object {
          "position": Array [
            500,
            65,
          ],
        },
      }
    }
  />
</ContextProvider>
`;
